dev/tests/proxy.rs

```diff
@@ -143,6 +209,62 @@ async fn simple_backend_proxy() -> Result<()> {
     Ok(())
 }
 
+#[integration_test]
+async fn simple_ws_backend_proxy() -> Result<()> {
+    let proxy = Proxy::new().await?;
+    let server = Server::serve_web_sockets().await?;
+
+    let sr = base_spawn_request();
+    proxy.db.insert_backend(&sr).await?;
+
+    proxy
+        .db
+        .insert_proxy_route(&sr.backend_id, "foobar", &server.address.to_string())
+        .await?;
+
+    let (ws_stream, ws_handshake_resp) = proxy
+        .https_websocket("foobar", "/")
+        .await
+        .expect("could not request ws from proxy");
+
+    assert_eq!(
+        ws_handshake_resp.status(),
+        http::status::StatusCode::SWITCHING_PROTOCOLS
+    );
+
+    let (mut write, read): (
+        SplitSink<WebSocketStream<TlsStream<TcpStream>>, Message>,
+        SplitStream<WebSocketStream<TlsStream<TcpStream>>>,
+    ) = StreamExt::split(ws_stream);
+
+    let max_messages = 6;
+    for i in 1..max_messages + 1 {
+        write
+            .send(Message::Text(format!("{}", i)))
+            .await
+            .expect("Failed to send message via proxy ws");
+    }
+
+    let read = tokio_stream::StreamExt::timeout(read, Duration::from_secs(10));
+
+    let responses: Vec<String> = read
+        .take(max_messages)
+        .map(|x| {
+            x.expect("response failed with timeout")
+                .expect("failure while reading from stream")
```

Comment:
Please run `cargo fmt` to clean up the indentation here. If it modifies files you haven't touched, you can revert them.